code401challenges/src/main/java/code401challenges/BinarySearch.java

```diff
@@ -6,19 +6,21 @@ public static void main(String[] args){
     System.out.println(binarySearch(inputArr, 15));
   }
 
-  public static int binarySearch(int[] sortedArr, int target){
-    int midIndex = sortedArr.length/2;
-    while(midIndex < sortedArr.length && midIndex > 0){
-      if(sortedArr[midIndex] == target){
-        return midIndex;
-      } else if(sortedArr[midIndex] < target){
-        midIndex = (midIndex + sortedArr.length)/2;
-      } else if(sortedArr[midIndex] > target){
-        midIndex = midIndex/2;
-      } else {
-        return -1;
+  public static int binarySearch(int[] arr, int target){
+    int minIndex = 0;
+    int maxIndex = arr.length - 1;
+    while(minIndex <= maxIndex ){
+      int guessIndex = (minIndex + maxIndex) / 2;
+      if(arr[guessIndex] == target){
```

Comment:
If you put these into one conditional using `if` and `else if` your code won't have to check every single conditional, it will instead stop as soon as it finds one that is `true`